### PR TITLE
Create default ProjectType/Stack/Template as part of setup

### LINKED ISSUE #1194: Add project type/template/stack stage to Setup process
### Epic

_No response_

### Description

**As a:** user installing FlowForge for the first time

**I want to:** be helped to create my first Project Type, Template and Stack

**So that:** I have to read less documentation and get a running Node-RED project as quickly as possible.

---

Our current install process requires an Admin to create a Project Type, Template and Stack before any projects can be created. We don't provide much help in doing that and expect a new user to quickly learn what those things are and what needs doing.

The Setup wizard should provide help so much of this can be automated with a minimum of input from the user.

1. Project Type - we can create a default Project Type with a suitable name/description.
2. Template - we can create a default template using all default values
3. Stack - this gets a *little* tricker as it will rely on the user doing *something* to get the stack into place. The wizard should be able to provide accurate instructions for the chosen driver. For localfs, that should be the appropriate set of commands to run (with all the correct paths set) that can be copy/pasted into a terminal. For docker/k8s, something else is needed - tbd.

### Which customers would this be availble to

All Users, (CE)

### Acceptance Criteria

- [ ] Project Type/Stack/Template are setup as part of the first-run setup wizard
- [ ] Install documentation updated to reflect the change
- [ ] First-run doc screenshots updated

### PR DESCRIPTION
**Pull request by knolleary**
Closes #1194 

To test this locally, https://github.com/flowforge/flowforge-driver-localfs/pull/71 will need to be merged first.

---

This creates a default ProjectType, Stack and Template as part of the setup wizard. The user isn't prompted for anything - they are just created.

If there is an error, they will get an error screen in the setup wizard and prompted to look in the log where we log the error details.

---

Before this gets merged, I will update the install docs as part of this PR to remove the relevant sections.

### CODE REVIEW

**Comment by knolleary** [comment]
Doc updates now added. This is good to review and merge once tests runs complete.

**Review by hardillb — APPROVED** [review]

**Comment by hardillb** [comment]
Only thought (which I'm capturing so I don't forget) is how this works on a totally fresh flowforge-dev-env checkout which won't have a default stack setup

**Comment by knolleary** [comment]
@hardillb re dev-env, good point. Will do a clean walk through of that tomorrow

File: docs/install/local/stacks.md [hunk 1]
```diff
@@ -11,47 +11,56 @@ For the Local deployment model, this covers two things:
     is useful when running on a memory constrained device such as a Raspberry Pi. Recommended minimum `256`.
  - `nodered` - the version number of Node-RED to use. This should match the value used in the steps following.
 
-## Upgrading Node-RED
+The FlowForge installer will create a default stack using the latest stable
+release of Node-RED.
 
-FlowForge supports Node-RED 2.2.x and later for creating stacks.
+The stacks are stored under `/opt/flowforge/var/stacks` or `c:\opt\flowforge\var\stacks` on Windows.
```
Comment (Steve-Mcl) [inline docs/install/local/stacks.md:17]:
```suggestion
The stacks are stored under `/opt/flowforge/var/stacks` or `c:\flowforge\var\stacks` on Windows.
```

Docs suggest install to `c:\flowforge`

File: docs/install/local/stacks.md [hunk 2]
```diff
@@ -11,47 +11,56 @@ For the Local deployment model, this covers two things:
...
+   ./bin/ff-install-stack.sh 3.0.2
+   ```
 
-Linux/Mac
-```
-bin/ff-install-stack.sh 3.0.2
-```
+   Windows
+   ```
+   cd c:\opt\flowforge
```
Comment (Steve-Mcl) [inline docs/install/local/stacks.md:39]:
```suggestion
   cd c:\flowforge
```

File: docs/install/local/stacks.md [hunk 3]
```diff
@@ -11,47 +11,56 @@ For the Local deployment model, this covers two things:
...
-   called `stacks`
-2. In the `var/stacks` directory create a directory called `3.0.2`
-3. In the `var/stacks/3.0.2` directory run `npm install --prefix . node-red@3.0.2`
-4. Create a new Stack under the Admin Settings section of the FlowForge web console.
-   1. Enter `3.0.2` under the Node-RED version - this *must* match the directory
-      name created under `var/stacks`.
+   Alternatively, click 'Create stack' to create an entirely new stack.
+
+   When prompted for the Node-RED version, provide the exact version that was installed. For example, if you ran the script with `latest` and it resulted in `3.0.2` being installed, you should enter `3.0.2`. This must
+   match the directory name created under `/opt/flowforge/var/stacks`.
```
Comment (Steve-Mcl) [inline docs/install/local/stacks.md:55]:
```suggestion
   match the directory name created in your `stacks` directory.
```

Suggestion: make this system agnostic?

**Pull request merged**